### PR TITLE
Get rid of deprecated maven-compat dependency

### LINKED ISSUE #127: Use Maven Resolver API instead of deprecated classes from maven-compat
To reduce the download footprint and to get rid of using deprecated classes (also for making this compatible with the upcoming Maven 4) the plugin should not depend on the deprecated maven-compat artifact (compare with https://issues.apache.org/jira/browse/MNG-7266? and https://cwiki.apache.org/confluence/display/MAVEN/Plugin+migration+to+Maven3+dependencies)

### PR DESCRIPTION
**Pull request by kwin**
Use Aether/Resolver API directly
This closes #127

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by kwin** [comment]
@cziegeler Can you review?

File: aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java [hunk 1]
```diff
@@ -237,10 +242,10 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             return;
         }
 
-        final VersionUtil versionUtil = new VersionUtil(this.getLog(), this.project, 
-                this.artifactHandlerManager, this.artifactMetadataSource, 
-                this.project.getRemoteArtifactRepositories(), this.project.getPluginArtifactRepositories(), 
-                this.mavenSession.getLocalRepository(), this.mavenSession.isOffline());
+        final VersionUtil versionUtil = new VersionUtil(this.getLog(), this.project, artifactHandlerManager,
+                this.repoSystem, this.repoSession,
+                this.remoteProjectRepositories, this.remotePluginRepositories,
```
Comment (cziegeler) [inline aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java:247]: Why can't we use project.getXXX ? This avoids the additional parameters

File: aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java [hunk 2]
```diff
@@ -558,27 +563,20 @@ private static Artifact findArtifact(final ArtifactId id, final Collection<Artif
...
      * @throws RuntimeException if the artifact can't be resolved
      */
-    @SuppressWarnings("deprecation")
     Artifact getOrResolveArtifact(final ArtifactId id) {
         Artifact result = this.artifactCache.get(id.toMvnId());
         if ( result == null ) {
             result = findArtifact(id, project.getAttachedArtifacts());
             if ( result == null ) {
-                result = findArtifact(id, project.getDependencyArtifacts());
+                result = findArtifact(id, project.getArtifacts());
```
Comment (cziegeler) [inline aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java:571]: getDependencyArtifacts lists (afaik) only the direct dependencies - while getArtifacts is transitive depending on the phase it gets invoked in. Maybe we need to switch to getDependencyList

Comment (kwin) [reply aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java:571]: I don't see a harm with also considering transitive dependencies, but maybe I am missing something. However, this is not related to the goal of this PR therefore I am reverting this change.

File: aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java (hunk 1 above)
Comment (kwin) [reply aemanalyser-maven-plugin/src/main/java/com/adobe/aem/analyser/mojos/AemAnalyseMojo.java]: Good point, done in https://github.com/adobe/aemanalyser-maven-plugin/pull/128/commits/f525d21c5fd4971bbd17370e1e9a01f7def77c30

**Review by cziegeler — APPROVED** [review]
Thanks, lgtm

**Pull request merged**